HousingResidentInformationAPI.Tests/V1/UseCase/ValidatePostcodeTests.cs

```diff
@@ -22,6 +22,11 @@ public void ValidPostcodesReturnTrue(string postcode)
         }
 
         [TestCase("1")]
+        [TestCase("111111")]
```

Comment:
Yes exactly. That's why I brought up this part of the Regex: `([Gg][Ii][Rr] 0[Aa]{2})|`. This part of the Regex makes this postcode match explicitly. You regex pretty much says: "Match **GIR 0AA** or the rest of UK postcodes"

Reply:
Is there a benefit to removing it?

Reply:
What is the benefit of removing it?

Reply:
Two benefits:
1) Cleaner code
2) Don't need to add test for it

Could also add that you get a micro performance benefit, because every postcode validation check would first check whether the patter is `GIR 0AA` or not. **Leaving this in is an equivalent of having this block of code at the start of your validator** _(roughly speaking)_:

```
if ( postcode.ToLower() == "gir 0aa" ) {
     return true;
}
```
So it's more a question to you - would you leave this block or delete it if you had recognized that it's redundant?

Of course, the performance benefit is would likely not be noticeable, so I won't bring this up as a 3rd reason.

Reply:
If this doesn't convince you, it's fine - leave it. I'm not demanding or anything to remove it :joy: The only reason I requested changes on this PR was because like I mentioned in the Review message - I think that this Regex needs higher test coverage.

Reply:
I only brought up GIR 0AA, because I thought it's odd to have it there, but this is PR Reviews - that's the entire point of them - to mark stuff up, bring it to attention and have a discussion on whether that should be changed or not. :)

Reply:
Like I stated, I prefer to leave the pattern unchanged.

Reply:
Unresolving because of this question:

> I'm still not sure what benefit there is in having an explicit testcase for gir oaa or GIR OAA

Since it's explicitly added, **adding a test for it increases test coverage** of the API, which is what the [README](https://github.com/LBHackney-IT/housing-resident-information-api/blob/master/README.md#testing) of this API implies we should do.

Reply:
The purpose of the PR was to increase the test coverage. The test coverage has been increased. It isn't exhaustive, which is also the case for the regex pattern, but the test coverage has objectively been increased.